api/src/main/java/org/ehcache/spi/serialization/Serializer.java

```diff
@@ -26,7 +26,20 @@
 
 /**
  * Interface defining the contract used to transform types in a serial form.
- * Implementations must be thread-safe and must define a constructor accepting a single ClassLoader parameter.
+ * 
+ * When used within the DefaultSerializationProvider which is the default implementation of {@link SerializationProvider} 
+ * service provided by ehcache, there are a few additional requirements:
+ * The implementations must be thread-safe and must define either or both of the two constructors:
```

Comment:
The thread-safe bit remains true always, so that should be indicated before the special integration with default serialization provider - which I would reference like in my sentence and not a class name given the module differences.